### ISSUE TITLE
False postive with GenericWhitespace on a "."

### ISSUE THREAD

**Issue by shague** [checkstyle/checkstyle#862]
The following code produces "warning: GenericWhitespace '>' is followed by whitespace." This is using checkstyle 6.1.1.
```
Map<UUID, TableUpdate<GenericTableSchema>.RowUpdate<GenericTableSchema>> rows;
```
The . in "a>.row" is causing the issue. Looks like the parser is expecting the next token to be "(" and anything else is reported as a space.

**Comment by damianszczepanik** [comment]
Very interesting construction, I will take a look at this.

**Comment by damianszczepanik** [comment]
@shague 
I checked given declaration and no violation was presented. I need you to check with recent checkstyle version since this rule was updated recently a few times. If you still have this problem please add:
- context for this declaration to see if this is class member or local variable
- configuration for this check to see which setting may have impact for this problem

**Comment by shague** [comment]
Damian,

I have attached the two files. TyperUtils.java is the file with the
warnings on lines 514,515 and 518. Recognize in the file I do have the
GenericWhitespace warnings suppressed for the method so just comment out
that line.

The checkstyle is used from maven-checkstyle-plugin v2.14. I also tried
with v2.15 and get the same result. This should be pulling in checkstyle
v6.1.1 so *I think* it is the latest. If not, how can I force the latest?

Thanks, Sam

On Sat, Mar 28, 2015 at 9:08 AM, Damian Szczepanik <notifications@github.com

**Comment by mkordas** [comment]
The latest version is 6.4.1. Please use the following config in Maven:
```xml
<plugin>
       <groupId>org.apache.maven.plugins</groupId>
       <artifactId>maven-checkstyle-plugin</artifactId>
       <version>2.15</version>
       <dependencies>
         <dependency>
            <groupId>com.puppycrawl.tools</groupId>
            <artifactId>checkstyle</artifactId>
            <version>6.4.1</version>
         </dependency>
       </dependencies>
        <configuration>
          ...
        </configuration>
     </plugin>       
```

**Comment by shague** [comment]
Excellent guys! This works perfectly. Thanks for teaching me something new
and for getting back to me so quickly.
